File: 324-wiggle-sort-ii/324-wiggle-sort-ii.cpp
class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        int n = nums.size();
        priority_queue<int> maxHeap(nums.begin(), nums.end());
        for(int i = 1; i < n; i += 2) {
            nums[i] = maxHeap.top();
            maxHeap.pop();
        }
        for(int i = 0; i < n; i += 2) {
            nums[i] = maxHeap.top();
            maxHeap.pop();
        }
    }
};

File: 324-wiggle-sort-ii/324-wiggle-sort-ii.py
class Solution:
    def wiggleSort(self, nums: List[int]) -> None:
        n, heap = len(nums), []
        for num in nums: 
            heappush(heap, -1 * num)
        for i in range (1, n, 2):
            nums[i] = -1 * heap[0]
            heappop(heap)
        for i in range (0, n, 2):
            nums[i] = -1 * heap[0]
            heappop(heap)